pkg/ingestor/parser/csaf/parser_csaf.go

```diff
@@ -290,36 +289,10 @@ func (c *csafParser) GetPredicates(ctx context.Context) *assembler.IngestPredica
 				if vi == nil {
 					continue
 				}
-
-				if status == "known_affected" || status == "under_investigation" {
-					vulnData := generated.ScanMetadataInput{
-						TimeScanned: c.csaf.Document.Tracking.CurrentReleaseDate,
```

Comment:
hmm why is this causing a duplication? Is it because of the timesacnned?

Reply:
No, it's because of the, further down, `CertifyVulnIngest` creation while only a `VexIngest` should be created.

Reply:
So this is expected. We could be in a situation where we have a VEX for a vulnerability that does not exist. That was the reason for the creation of the  `CertifyVulnIngest`. It is probably being duplicated because of the `TimeScanned` are not the same between the two. We are doing a similar thing for OpenVEX and CDX Vex.

Reply:
This is to ensure that we have a `certifyVuln` node associated with a VEX at all times. `certifyVuln` nodes can have duplicates (based on the data source they came from or from a timescanned). The certifier also takes in an input for `daysSinceLastScan` and in that case, you could have duplicates.

Reply:
Thoughts on the other maintainers? @lumjjb, @mihaimaruseac, @jeffmendoza

Reply:
> This is to ensure that we have a `certifyVuln` node associated with a VEX at all times. `certifyVuln` nodes can have duplicates (based on the data source they came from or from a timescanned). The certifier also takes in an input for `daysSinceLastScan` and in that case, you could have duplicates.

You lost me.
We have have both `CertifyVEXStatement` and `CertifyVuln` so we haven't a `CertifyVuln` node associated with a VEX all the times.
Am I missing/misunderstanding something?

Reply:
So I am thinking about it from a query standpoint. We are looking for `CertifyVuln` nodes to determine if a package has a vulnerability. If, for whichever reason, the package has a `CertifyVEXStatement` node attached to a vulnerability and package but the same package does not have a `CertifyVuln` node, would that cause issues in the query.

Reply:
@lumjjb @jeffmendoza wanted to get your thoughts on this

Reply:
A VEX statement, on its own, is a round-about assertion that the package is affected by the vuln. Creating a `CertifyVuln` in this case is appropriate. The `Justification` should clarify that it was found in a VEX statement

Reply:
> A VEX statement, on its own, is a round-about assertion that the package is affected by the vuln. Creating a `CertifyVuln` in this case is appropriate. The `Justification` should clarify that it was found in a VEX statement

I know what you mean but in the usage we did of the VEX statement it makes the `CertifyVuln` "useless". In fact `CertifyVEXStatement` has more information than `CertifyVuln` and it doesn't just say if a package is affected but also if it's not affected, fixed or still under investigation.
So from a user perspective, the `CertifyVEXStatement` is the first thing you want to know and having also a `CertifyVuln` attached for the same vulnerability to the same package is not adding any value to the final user's knowledge.

So guac can keep the "double" ingestion but the `CertifyVuln` looks like a subset of the  `CertifyVEXStatement` information adding no real value but that doubles the "cost" of ingestion and that's the reason behind this PR.

Reply:
My thoughts are that VEX intention is overloaded today. As a concept VEX acts as a counter assertion of CertifyVuln. However, today, folks are using it as a way to be a vulnerability disclosure report. This is a manifestation of convenience unfortunately. So VEX as a concept should not certify if something is vulnerable. VEX statements as a data format unfortunately does...

My thoughts are that it is fine to have the duplicate of verbs for now, but eventually we want to have a more proper split of the concept vs data format, alike to "Build provenance vs SLSA" and "Software composition vs SBOM"

Reply:
Thanks a lot to everyone for the contributions.
Let me say I'm happy to close this PR because it looks like having both is fine.